csharp/src/Google.Protobuf/Reflection/FileDescriptor.cs

```diff
@@ -240,6 +243,11 @@ private static IList<FileDescriptor> DeterminePublicDependencies(FileDescriptor
         /// </value>
         public IList<ServiceDescriptor> Services { get; }
 
+        /// <summary>
+        /// Unmodifiable list of top-level extensions declared in this file.
+        /// </summary>
+        public ExtensionCollection Extensions { get; }
```

Comment:
Can't  use IList<FieldDescriptor>?

Reply:
We could, however with it fashioned in this way we can provide methods to get fields defined for specific messages in declaration or number order (which I believe will be a common use-case like in FieldDescriptor). If consumers want extensions in any order not separated by extended type, they can get `Extensions.UnorderedExtensions` (which we can rename to just `Unordered` if it's too repetitive).

Reply:
ExtesnionCollection make sense as we also have FieldCollection.